File: src/main/java/eotree/data/EOStringData.java
package eotree.data;

/**
 * EBNF representation:
 * <code>
 * /"[^"]*"/
 * </code>
 */
public class EOStringData extends EOData {
    public String str;

    public EOStringData(String str) {
        this.str = str;
    }

    @Override
    public String generateEO(int indent) {
        return "\"" + str + "\"";
    }
}

File: src/main/java/eotree/data/EOStringData.kt
package eotree.data

/**
 * EBNF representation:
 * `
 * /"[^"]*"/
` *
 */
class EOStringData(var str: String) : EOData() {
    override fun generateEO(indent: Int): String {
        return indent(indent) + "\"" + str + "\""
    }
}
